### DISCUSSION TITLE
How to automatically fill/generate the template with pictures files ? needed for the printing house/printshop

### DISCUSSION THREAD

**Discussion by szplukasz** [ImageMagick/ImageMagick#2714] [General]
At first, I apologize for the poor language
I have a lot of pictures from many sources
at my printing house they gave me a A4 template (pdf file) to prepare project of gaming cards for printing and automatic cutting ( bleeds , cut lines)
is there a way to auto-load multiple image files, rescale, fit this into template and generate output pdf files ? There are 9 pictures per A4/page.
PS 1 maybe another way to Automatically Resize, Fit, and Align Images into mentioned template (generate template) ?
Repeatability, autoadding a cutting line and bleeds is required .
PS 2 Please Look at picture to better understand my problem
https://imgur.com/1OtIfNU
https://imgur.com/RE8CizJ
Print template:
http://piotrnowinski.pl/wp-content/uploads/2014/02/tutorial-trade-cards-print-template.pdf

Thank you for your help

**Comment by snibgo** [comment]
ImageMagick is a raster image processor. (Raster means images with pixels, not vectors.) It can rasterize your PDF, and combine with with other raster images to make a single raster image. Then it can write a PDF file that contains that raster image.

I'm not sure if that is what you want.

**Comment by szplukasz** [comment]
Great. is it possible to make the source images automatically
processed to match the template pattern?


wt., 13 paź 2020, 23:36 użytkownik snibgo <notifications@github.com>
napisał:

**Comment by snibgo** [comment]
Each image can be cropped or resized or both, if you know the required dimensions.

You would get the required dimensions, and the required positions of the nine sources, from the template. You could do this manually. If the template keeps changing, you might do this automatically.
